### PR TITLE
feat!: add map editor

### PR DESCRIPTION
**Pull request by jamestansx**
Okay, this is quite a huge PR tbh.. but it's fine, here are a list of changes that I have made:

- refactor the map rendering mechanism. Use file to handle the map storage. It will be stored in `%APPDATA%` folder.
- refactor the key input event in main menu. Use `kbhit()` to detect keystroke instead of `GetAsyncKeyState` bcoz it causes too much problems...
- feat!!: introduce map editor. Added `new` and `edit` feature. Players can add new maps from existing maps as the template. `Notepad` is used for editing maps (who cares).. 

There are some issues that need to follow up:
- Change the enemy and player rendering/detection mechanism. Incorporate it with the new expanded map
- Control the console size to avoid overflow the visible screen.

### CODE REVIEW

File: game.hpp [hunk 1]
```diff
@@ -30,12 +40,12 @@ void delete_old_position(short x, short y) {
 }
 
 bool check_obstacles(short x, short y) {
-    return map[y][x] == '#' || map[y][x] == '|';
+    return map[y][x] == '-' || map[y][x] == '|';
```
Comment (Steveo-O) [inline game.hpp:43]: after this change from '#' to '|' still can detect the obstacles?

Comment (jamestansx) [reply game.hpp:43]: Oh damn.. I actually just wanna try out if using `-` would make the map looks less complicated. 

Do u wanna change to `-`?

**Comment by jamestansx** [comment]
Btw, do we need to ship default maps to users? If so, do u prefer to copy the maps during compilation time, or embed the maps directly in source code?

File: game.hpp (hunk 1 above)
Comment (Steveo-O) [reply game.hpp:43]: That will make the map weird bcz the player cannot run to the extra space, make confusion?

**Pull request merged**